policy/src/main/java/org/apache/brooklyn/policy/ha/ServiceFailureDetector.java

```diff
@@ -230,6 +237,7 @@ protected void setActualState(Maybe<Lifecycle> state) {
                     }
                     lastPublished = LastPublished.FAILED;
                     entity.sensors().emit(HASensors.ENTITY_FAILED, new HASensors.FailureDescriptor(entity, getFailureDescription(now)));
+                    requestPersist();
```

Comment:
Oh that's sneaky. Can you use config instead? (not sure if config change triggers persistence on adjuncts).